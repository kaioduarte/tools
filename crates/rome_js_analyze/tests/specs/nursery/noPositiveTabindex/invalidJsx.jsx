let test;

test = <div tabIndex={1} />;

test = <div tabIndex={"1"} />;

test = <div tabIndex={'5'} />;

// not working - inner_string_text can't capture the number
test = <div tabIndex="1" />;

test = <div tabIndex={1}>foo</div>;

test = <div tabIndex={"1"}>foo</div>;

test = <div tabIndex={'5'}>foo</div>;

test = <div tabIndex="1">foo</div>;
