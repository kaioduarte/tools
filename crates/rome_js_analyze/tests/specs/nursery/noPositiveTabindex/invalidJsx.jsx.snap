---
source: crates/rome_js_analyze/tests/spec_tests.rs
assertion_line: 100
expression: invalidJsx.jsx
---
# Input
```js
<>
	<div tabIndex={1} />
	<div tabIndex={"1"} />
	<div tabIndex={'5'} />
	<div tabIndex="1" />
	<div tabIndex={1}>foo</div>
	<div tabIndex={"1"}>foo</div>
	<div tabIndex={'5'}>foo</div>
	<div tabIndex={+5}>foo</div>
	<div tabIndex={+05}>foo</div>
	<div tabIndex="1">foo</div>
</>

```

# Diagnostics
```
invalidJsx.jsx:2:7 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
    1 │ <>
  > 2 │ 	<div tabIndex={1} />
      │ 	     ^^^^^^^^
    3 │ 	<div tabIndex={"1"} />
    4 │ 	<div tabIndex={'5'} />
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```

```
invalidJsx.jsx:3:7 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
    1 │ <>
    2 │ 	<div tabIndex={1} />
  > 3 │ 	<div tabIndex={"1"} />
      │ 	     ^^^^^^^^
    4 │ 	<div tabIndex={'5'} />
    5 │ 	<div tabIndex="1" />
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```

```
invalidJsx.jsx:4:7 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
    2 │ 	<div tabIndex={1} />
    3 │ 	<div tabIndex={"1"} />
  > 4 │ 	<div tabIndex={'5'} />
      │ 	     ^^^^^^^^
    5 │ 	<div tabIndex="1" />
    6 │ 	<div tabIndex={1}>foo</div>
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```

```
invalidJsx.jsx:5:7 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
    3 │ 	<div tabIndex={"1"} />
    4 │ 	<div tabIndex={'5'} />
  > 5 │ 	<div tabIndex="1" />
      │ 	     ^^^^^^^^
    6 │ 	<div tabIndex={1}>foo</div>
    7 │ 	<div tabIndex={"1"}>foo</div>
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```

```
invalidJsx.jsx:6:7 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
    4 │ 	<div tabIndex={'5'} />
    5 │ 	<div tabIndex="1" />
  > 6 │ 	<div tabIndex={1}>foo</div>
      │ 	     ^^^^^^^^
    7 │ 	<div tabIndex={"1"}>foo</div>
    8 │ 	<div tabIndex={'5'}>foo</div>
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```

```
invalidJsx.jsx:7:7 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
    5 │ 	<div tabIndex="1" />
    6 │ 	<div tabIndex={1}>foo</div>
  > 7 │ 	<div tabIndex={"1"}>foo</div>
      │ 	     ^^^^^^^^
    8 │ 	<div tabIndex={'5'}>foo</div>
    9 │ 	<div tabIndex={+5}>foo</div>
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```

```
invalidJsx.jsx:8:7 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
     6 │ 	<div tabIndex={1}>foo</div>
     7 │ 	<div tabIndex={"1"}>foo</div>
   > 8 │ 	<div tabIndex={'5'}>foo</div>
       │ 	     ^^^^^^^^
     9 │ 	<div tabIndex={+5}>foo</div>
    10 │ 	<div tabIndex={+05}>foo</div>
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```

```
invalidJsx.jsx:9:7 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
     7 │ 	<div tabIndex={"1"}>foo</div>
     8 │ 	<div tabIndex={'5'}>foo</div>
   > 9 │ 	<div tabIndex={+5}>foo</div>
       │ 	     ^^^^^^^^
    10 │ 	<div tabIndex={+05}>foo</div>
    11 │ 	<div tabIndex="1">foo</div>
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```

```
invalidJsx.jsx:10:7 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
     8 │ 	<div tabIndex={'5'}>foo</div>
     9 │ 	<div tabIndex={+5}>foo</div>
  > 10 │ 	<div tabIndex={+05}>foo</div>
       │ 	     ^^^^^^^^
    11 │ 	<div tabIndex="1">foo</div>
    12 │ </>
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```

```
invalidJsx.jsx:11:7 lint/nursery/noPositiveTabindex ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Avoid positive values for the tabIndex prop.
  
     9 │ 	<div tabIndex={+5}>foo</div>
    10 │ 	<div tabIndex={+05}>foo</div>
  > 11 │ 	<div tabIndex="1">foo</div>
       │ 	     ^^^^^^^^
    12 │ </>
    13 │ 
  
  i Elements with a positive tab index override natural page content order. This causes elements without a positive tab index to come last when navigating using a keyboard.
  

```


